### PR TITLE
Support for multiple objects

### PR DESCRIPTION
**Pull request by tomasmarciulionis**

### CODE REVIEW

File: android/app/src/androidTest/java/com/trafi/example/ButtonPreviews.kt [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
-import com.trafi.ui.theme.MaasCornerRadius
-import com.trafi.ui.theme.MaasTheme
 import org.junit.Before
 import org.junit.Rule
 import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+
+@RunWith(Parameterized::class)
+class ComposablePreviewsTests(private val preview: @Composable () -> Unit): ScreenshotTest {
```
Comment (justasm) [inline android/app/src/androidTest/java/com/trafi/example/ButtonPreviews.kt:14]: File should be renamed, too

File: android/app/src/androidTest/java/com/trafi/example/TestPreviews.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.unit.dp
+import com.trafi.ui.Button
+import com.trafi.ui.ButtonPreview
+import com.trafi.ui.ButtonStyledCornerRadiusPreview
+import com.trafi.ui.theme.MaasCornerRadius
+import com.trafi.ui.theme.MaasTheme
+
+val testPreviews = listOf<@Composable () -> Unit> (
```
Comment (justasm) [inline android/app/src/androidTest/java/com/trafi/example/TestPreviews.kt:11]: Does this work? 😮 Cool!

**Review by justasm — COMMENTED** [review]
Badge.3x.png appeared again 🙈 And please rebase on the latest `main` to avoid the funky git commit history.

File: android/app/src/androidTest/java/com/trafi/example/TestPreviews.kt (hunk 2 above)
Comment (tomasmarciulionis) [reply android/app/src/androidTest/java/com/trafi/example/TestPreviews.kt:11]: Yes, this works. :)

File: android/app/src/androidTest/java/com/trafi/example/ButtonPreviews.kt (hunk 1 above)
Comment (tomasmarciulionis) [reply android/app/src/androidTest/java/com/trafi/example/ButtonPreviews.kt:14]: Done

**Review by justasm — APPROVED** [review]

File: android/app/src/androidTest/java/com/trafi/example/TestPreviews.kt [hunk 3]
```diff
@@ -1,18 +1,10 @@
...
-val testPreviews = listOf<@Composable () -> Unit> (
-    {
-        ButtonPreview()
-    },
-    {
-        ButtonStyledCornerRadiusPreview()
-    }
+val testPreviews = listOf(
+        arrayOf<Pair<@Composable () -> Unit, String>>(Pair({ButtonPreview()}, "ButtonPreview")),
+        arrayOf<Pair<@Composable () -> Unit, String>>(Pair({ButtonStyledCornerRadiusPreview()}, "ButtonStyledCornerRadiusPreview"))
```
Comment (justasm) [inline android/app/src/androidTest/java/com/trafi/example/TestPreviews.kt:9]: Can we write it like this..

```kotlin
val testPreviews: List<Array<Pair<@Composable () -> Unit, String>>> = listOf(
    arrayOf({ ButtonPreview() } to "ButtonPreview"),
    arrayOf({ ButtonStyledCornerRadiusPreview() } to "ButtonStyledCornerRadiusPreview"),
)
```

and what is the purpose of wrapping the `Pair` in an `Array` type?

File: android/app/src/androidTest/java/com/trafi/example/ComposablePreviewsTests.kt [hunk 4]
```diff
@@ -23,12 +29,10 @@ class ButtonPreviews: ScreenshotTest {
...
 
     @Test
-    fun buttonTest() {
+    fun previewTest() {
         previewTest
-            .renderAndTestView {
-                MaasTheme(cornerRadius = MaasCornerRadius(buttonRadius = 0.dp)) {
-                    Button("Unlock now", onClick = {})
-                }
+            .renderAndTestView(screenshotName = previewWithName.second) {
```
Comment (justasm) [inline android/app/src/androidTest/java/com/trafi/example/ComposablePreviewsTests.kt:34]: Might be nice to decompose first, i.e.

```kotlin
val (name, preview) = previewWithName
// ..
  .renderAndTestView(name) { preview() }
```

File: android/app/src/androidTest/java/com/trafi/example/TestPreviews.kt (hunk 3 above)
Comment (tomasmarciulionis) [reply android/app/src/androidTest/java/com/trafi/example/TestPreviews.kt:9]: I also tinkered with it a bit, and such case works: 
`package com.trafi.example

import androidx.compose.runtime.Composable
import com.trafi.ui.ButtonPreview
import com.trafi.ui.ButtonStyledCornerRadiusPreview

@Suppress("RemoveExplicitTypeArguments")
val testPreviews = listOf<Pair<@Composable () -> Unit, String>>(
        Pair({ ButtonPreview() }, "ButtonPreview"),
        Pair({ ButtonStyledCornerRadiusPreview() }, "ButtonStyledCornerRadiusPreview")
)`

Comment (tomasmarciulionis) [reply android/app/src/androidTest/java/com/trafi/example/TestPreviews.kt]: Somehow the `({ ButtonPreview() } to "ButtonPreview")` doesn't work as it throws static code analysis error.

**Pull request merged**